### PR TITLE
208 visualizatize spaces and improve relations

### PR DESCRIPTION
**Pull request by DaJansenGit**
This brings:
* [x] Improvements in the code of the task `CombineThermalZones`
* [X] An visualization function that uses OCC to display how the `ThermalZone` elements are aggregated together
* [X] An example for the visualization
* [X] Improvements in the way how relations between elements (e.g. `Storey` <-> `Building`) etc. are handeld and stored.

The visualization function creates images like attached for different zoning strategies (example is DigitalHub building)

zoning_all_criteria
zoning_external
zoning_external_orientation
zoning_external_orientation_usage
zoning_usage

### CODE REVIEW

File: bim2sim/tasks/bps/bind_tz.py [hunk 1]
```diff
@@ -17,101 +17,87 @@ class CombineThermalZones(ITask):
         * Usage
         * Window to wall ratio
     """
-
```
Comment (veronikarichter) [inline bim2sim/tasks/bps/bind_tz.py:20]: please rename the file bind_tz to something more related to combining thermal zones (bind_tz sounds like you're binding the thermal zones to the building, but does not sound like aggregation).

File: bim2sim/tasks/bps/bind_tz.py [hunk 2]
```diff
@@ -17,101 +17,87 @@ class CombineThermalZones(ITask):
...
             self.logger.info("Reduced number of thermal zones from %d to  %d",
-                             n_zones_before, len(self.bounded_tz))
-        self.add_storeys_to_buildings(elements)
+                             n_zones_before, len(tz_elements_after))
 
-        return self.bounded_tz,
-
-    def bind_tz_one_zone(self, thermal_zones, elements):
+    @staticmethod
+    def bind_tz_one_zone(thermal_zones, elements):
```
Comment (veronikarichter) [inline bim2sim/tasks/bps/bind_tz.py:40]: same here, please refactor these methods to something like "combine" / "aggregate" / ...

Comment (veronikarichter) [reply bim2sim/tasks/bps/bind_tz.py:40]: Same for following methods

File: bim2sim/plugins/PluginTEASER/bim2sim_teaser/utilities/visualize_spaces.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
+"""
```
Comment (veronikarichter) [inline bim2sim/plugins/PluginTEASER/bim2sim_teaser/utilities/visualize_spaces.py:1]: please move/keep this file to/in the original bim2sim/utilities/visualize_spaces.py, so we can add more plot functions afterwards from other Plugins (e.g., PluginEnergyPlus utils)

File: bim2sim/plugins/PluginTEASER/bim2sim_teaser/utilities/visualize_spaces.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+         rgb: must be a tuple with 3 values [0,1]. e.g. (0, 0.5, 0.7)
+
+    Returns:
+        Quantity_Color object which is compatible with with the OCC viewer.
+    """
+    return Quantity_Color(rgb[0], rgb[1], rgb[2], Quantity_TOC_RGB)
+
+
+def visualize_zones(
+        thermal_zones: list, path: Path, filename: Union[str, None] = None):
```
Comment (veronikarichter) [inline bim2sim/plugins/PluginTEASER/bim2sim_teaser/utilities/visualize_spaces.py:50]: please specify that thermal_zones is a list of bim2sim thermal_zones.

File: bim2sim/tasks/bps/bind_tz.py [hunk 5]
```diff
@@ -17,101 +17,87 @@ class CombineThermalZones(ITask):
...
         """groups together all the thermal zones as one building"""
         tz_group = {'one_zone_building': thermal_zones}
         new_aggregations = AggregatedThermalZone.find_matches(
             tz_group, elements)
-        for inst in new_aggregations:
-            self.bounded_tz.append(inst)
 
-    def bind_tz_criteria(self, elements):
+    def bind_tz_criteria(self, thermal_zones: list, elements: dict):
```
Comment (veronikarichter) [inline bim2sim/tasks/bps/bind_tz.py:46]: rename

File: bim2sim/tasks/common/create_relations.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    ThermalZone, Storey, Building, ExternalSpatialElement)
+from bim2sim.tasks.base import ITask
+from bim2sim.elements.mapping.ifc2python import getBuilding, getStorey, getSite
+
+
+class CreateRelations(ITask):
+    """Relations of elements, run() method holds detailed information."""
+    reads = ('elements',)
+
+    def run(self, elements: dict):
```
Comment (veronikarichter) [inline bim2sim/tasks/common/create_relations.py:11]: please specify the dicitonary content (ProductBased, IfcBased, ...)

File: bim2sim/utilities/visualize_spaces.py [hunk 7]
```diff
@@ -1,45 +0,0 @@
-import ifcopenshell
```
Comment (veronikarichter) [inline bim2sim/utilities/visualize_spaces.py:1]: keep this file and replace with plot function which you just tried to move to TEASER Plugin.

**Comment by DaJansenGit** [comment]
@veronikarichter: Addressed all requested changes, please review again.

**Review by veronikarichter — APPROVED** [review]

**Pull request merged**